src/Microsoft.VisualBasic/tests/VB/SpecialDirectoriesTests.vb

```diff
@@ -0,0 +1,99 @@
+                Assert.True(Env_Desktop = FileIO_Desktop, $"{Env_Desktop} <> {FileIO_Desktop}")
+            End If
+        End Sub
+
+        <Fact>
+        Public Shared Sub MyDocumentsFolderTest()
+            If PlatformDetection.IsWindowsNanoServer Then
+                Exit Sub
+            End If
+
+            If Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments).Length = 0 Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.MyDocuments)
+            Else
+                Assert.Equal(Environment.GetFolderPath(SpecialFolder.Personal).TrimEnd(Separators), SpecialDirectories.MyDocuments.TrimEnd(Separators))
+            End If
+        End Sub
+
+        <PlatformSpecific(TestPlatforms.Windows), ConditionalFact(GetType(PlatformDetection), NameOf(PlatformDetection.IsWindowsNanoServer))>
+        Public Shared Sub MyDocumentsFolderTestForNano()
+            Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.MyDocuments)
+        End Sub
+
+
+        <Fact>
+        Public Shared Sub MyMusicFolderTest()
+            If Environment.GetFolderPath(Environment.SpecialFolder.MyMusic).Length = 0 Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.MyMusic)
+            Else
+                Assert.Equal(Environment.GetFolderPath(SpecialFolder.MyMusic).TrimEnd(Separators), SpecialDirectories.MyMusic.TrimEnd(Separators))
+            End If
+        End Sub
+
+        <Fact>
+        Public Shared Sub MyPicturesFolderTest()
+            If Environment.GetFolderPath(Environment.SpecialFolder.MyPictures).Length = 0 Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.MyPictures)
+            Else
+                Assert.Equal(Environment.GetFolderPath(SpecialFolder.MyPictures).TrimEnd(Separators), SpecialDirectories.MyPictures.TrimEnd(Separators))
+            End If
+        End Sub
+
+        <Fact>
+        Public Shared Sub ProgramFilesFolderTest()
+            If Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles).Length = 0 Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.ProgramFiles)
+            Else
+                Assert.Equal(Environment.GetFolderPath(SpecialFolder.ProgramFiles).TrimEnd(Separators), SpecialDirectories.ProgramFiles.TrimEnd(Separators))
+            End If
+        End Sub
+
+        <Fact>
+        Public Shared Sub ProgramsFolderTest()
+            If Environment.GetFolderPath(Environment.SpecialFolder.Programs).Length = 0 Then
+                Assert.Throws(Of IO.DirectoryNotFoundException)(Function() SpecialDirectories.Programs)
+            Else
+                Assert.Equal(Environment.GetFolderPath(SpecialFolder.Programs).TrimEnd(Separators), SpecialDirectories.Programs.TrimEnd(Separators))
+            End If
+        End Sub
```

Comment:
The `SpecialFolder` tests look similar. Consider extracting a helper method that can be used for each: `CheckSpecialFolder(folder As SpecialFolder, getSpecialDirectory As Func(Of String))`.

Reply:
Done, great suggestion.